pymeasure/display/console.py

```diff
@@ -0,0 +1,276 @@
+    :param log_level: logging level
+    :param sequence_file: simple text file to quickly load a pre-defined
+    sequence with the :code:`Load sequence` button
+    :param directory_input: specify, if present, where the experiment's result
+    will be saved.
+    """
+    def __init__(self,
+                 args,
+                 procedure_class,
+                 inputs=(),
+                 log_channel='',
+                 log_level=logging.INFO,
+                 sequence_file=None,
+                 directory_input=False,
+                 ):
+
+        super().__init__(args)
+        self.args = args
+        self.procedure_class = procedure_class
+        self.inputs = inputs
+        self.sequence_file = sequence_file
+        self.directory_input = directory_input
+        self.log = logging.getLogger(log_channel)
+        self.log_level = log_level
+        log.setLevel(log_level)
+        self.log.setLevel(log_level)
+
+        # Check if the get_estimates function is reimplemented
+        self.use_estimator = not self.procedure_class.get_estimates == Procedure.get_estimates
+        self.parser = ConsoleArgumentParser(procedure_class)
+        self.parser.setup_parser(self.inputs)
+        if self.use_estimator:
+            log.warning("Estimator not yet implemented")
+        # Handle Ctrl+C nicely
+        signal.signal(signal.SIGINT, lambda sig, _: self.abort())
+
+    def get_filename(self, directory):
+        """ Return filename for logging.
+
+        User can override this method to define their own filename
+        """
+        if self.filename is not None:
+            return os.path.join(directory, self.filename)
+        else:
+            return unique_filename(directory)
+
+    def _update_progress(self, progress):
+        if self.bar:
+            self.bar.update(progress)
+
+    def _update_status(self, status):
+        if self.bar:
+            self.bar.update(status=Procedure.STATUS_STRINGS[status])
+
+    def _update_log(self, record):
+        log.emit(record)
+
+    def _failed(self):
+        self._terminate("Manager's running experiment has failed")
+
+    def _abort_returned(self):
+        self._terminate("Running experiment has returned after an abort")
+
+    def _finish(self):
+        self._terminate("Running experiment has finished", 100.0)
+
+    def _terminate(self, debug_message, update_bar=None):
+        log.debug(debug_message)
+        self._monitor.wait()
+        log.debug("Monitor has cleaned up after the Worker")
+        if self.bar:
+            self.bar.update(update_bar)
+            self.bar.finish()
+        self.quit()
+
+    def abort(self):
+        """ Aborts the currently running Experiment, but raises an exception if
+        there is no running experiment
+        """
+        self._worker.update_status(Procedure.ABORTED)
+        self._worker.stop()
+
+    def exec(self):
+        # Parse command line arguments
+        args = vars(self.parser.parse_args(self.args[1:]))
```

Comment:
I think the `self.args[1:]` is not required; it should normally automatically handle the sys.argv correctly

Reply:
Ok. I will make the change

Reply:
Done